kernel/kernel-defaults/src/test/scala/io/delta/kernel/defaults/DeltaTableSchemaEvolutionSuite.scala

```diff
@@ -0,0 +1,1212 @@
+        .add("a", StringType.STRING, true)
+        .add(
+          "map",
+          new MapType(
+            StringType.STRING,
+            new ArrayType(
+              new StructType().add("nested_map_value", IntegerType.INTEGER),
+              true),
+            false))
+
+      createEmptyTable(
+        engine,
+        tablePath,
+        initialSchema,
+        tableProperties = Map(
+          TableConfig.COLUMN_MAPPING_MODE.getKey -> "name",
+          TableConfig.ICEBERG_COMPAT_V2_ENABLED.getKey -> "true"))
+
+      val currentSchema = table.getLatestSnapshot(engine).getSchema
+      val mapSchema = currentSchema.get("map").getDataType.asInstanceOf[MapType]
+      val arrayValue = mapSchema.getValueType.asInstanceOf[ArrayType]
+      val innerStruct = arrayValue.getElementType.asInstanceOf[StructType]
+      val newSchema = new StructType()
+        .add("a", StringType.STRING, true, currentSchema.get("a").getMetadata)
+        .add(
+          "map",
+          new MapType(
+            StringType.STRING,
+            new ArrayType(
+              new StructType().add(
+                "nested_map_value",
+                IntegerType.INTEGER,
+                innerStruct.get("nested_map_value").getMetadata)
+                .add(
+                  "new_required_field",
+                  IntegerType.INTEGER,
+                  false,
+                  fieldMetadataForColumn(7, "7")),
+              true),
+            false),
+          true,
+          fieldMetadataForMapColumn(4, "map", "map", 5, 6))
+
+      assertSchemaEvolutionFails[KernelException](
+        table,
+        engine,
+        newSchema,
+        "Cannot add non-nullable field new_required_field")
+    }
+  }
+
+  test("Cannot drop a partition column") {
+    withTempDirAndEngine { (tablePath, engine) =>
+      val table = Table.forPath(engine, tablePath)
+      val initialSchema = new StructType()
+        .add("a", StringType.STRING, true)
+        .add("c", IntegerType.INTEGER, true)
+
+      createEmptyTable(
+        engine,
+        tablePath,
+        initialSchema,
+        partCols = Seq("c"),
+        tableProperties = Map(
+          TableConfig.COLUMN_MAPPING_MODE.getKey -> "id",
+          TableConfig.ICEBERG_COMPAT_V2_ENABLED.getKey -> "true"))
+
+      val currentSchema = table.getLatestSnapshot(engine).getSchema
+      val newSchema = new StructType()
+        .add("a", StringType.STRING, true, currentSchema.get("a").getMetadata)
+        .add(
+          "b",
+          new StructType()
+            .add("d", IntegerType.INTEGER, true, fieldMetadataForColumn(4, "d"))
+            .add("e", IntegerType.INTEGER, true, fieldMetadataForColumn(5, "e")),
+          true,
+          fieldMetadataForColumn(3, "b"))
+
+      assertSchemaEvolutionFails[IllegalArgumentException](
+        table,
+        engine,
+        newSchema,
+        "Partition column c not found in the schema")
+    }
+  }
+
+  test("Cannot rename a partition column") {
```

Comment:
I actually thought for some reason we thought this was supported? Haha I can't remember at this point

Reply:
Or "should be" obviously we can not support it for now